client/app/(auth)/createProfile.tsx

```diff
@@ -1,19 +1,430 @@
-import { View, Text } from "react-native";
-import { useAppSelector } from "@/state/hooks";
+import { yupResolver } from "@hookform/resolvers/yup";
+import DateTimePicker from "@react-native-community/datetimepicker";
+import { getAuth } from "@react-native-firebase/auth";
+import { router } from "expo-router";
+import { WarningCircle } from "phosphor-react-native";
+import { useEffect, useState } from "react";
+import { useForm, Controller } from "react-hook-form";
+import {
+  Image,
+  View,
+  KeyboardAvoidingView,
+  Text,
+  TextInput,
+  TouchableWithoutFeedback,
+  Keyboard,
+  ImageSourcePropType,
+  Modal,
+  Pressable,
+} from "react-native";
+import * as yup from "yup";
+import BottomSheet from "@/components/BottomSheet";
+import Spinner from "@/components/Spinner";
+import { UserController, UserEditModel } from "@/controllers/userController";
+import { useAppDispatch } from "@/state/hooks";
+import { setUser } from "@/state/slices/userSlice";
+
+type AvatarOption = {
+  url: string;
+  source: ImageSourcePropType;
+};
+
+const avatarOptions: AvatarOption[] = [
+  {
+    url: "../../assets/images/placeholder-avatar.png",
+    source: require("../../assets/images/placeholder-avatar.png"),
+  },
+  {
+    url: "../../assets/images/profile/profile-1.png",
+    source: require("../../assets/images/profile/profile-1.png"),
+  },
+  {
+    url: "../../assets/images/profile/profile-2.png",
+    source: require("../../assets/images/profile/profile-2.png"),
+  },
+  {
+    url: "../../assets/images/profile/profile-3.png",
+    source: require("../../assets/images/profile/profile-3.png"),
+  },
+  {
+    url: "../../assets/images/profile/profile-4.png",
+    source: require("../../assets/images/profile/profile-4.png"),
+  },
+  {
+    url: "../../assets/images/profile/profile-5.png",
+    source: require("../../assets/images/profile/profile-5.png"),
+  },
+  {
+    url: "../../assets/images/profile/profile-6.png",
+    source: require("../../assets/images/profile/profile-6.png"),
+  },
+  {
+    url: "../../assets/images/profile/profile-7.png",
+    source: require("../../assets/images/profile/profile-7.png"),
+  },
+  {
+    url: "../../assets/images/profile/profile-8.png",
+    source: require("../../assets/images/profile/profile-8.png"),
+  },
+  {
+    url: "../../assets/images/profile/profile-9.png",
+    source: require("../../assets/images/profile/profile-9.png"),
+  },
+];
+
+const schema = yup.object({
+  avatarUrl: yup
+    .string()
+    .required("Avatar is required")
+    .test(
+      "not-default-avatar",
+      "Avatar is required",
+      (value) => value !== avatarOptions[0].url,
+    ),
+  dateOfBirth: yup.date().required("Date of Birth is required"),
+  username: yup.string().required("Username is required"),
+  name: yup.string().required("Name is required"),
+});
 
 const CreateProfile = () => {
-  const user = useAppSelector((state) => state.user);
+  const auth = getAuth();
+  const currentUser = auth.currentUser;
```

Comment:
I think fetching for the id from the user slice is more fitting since we that is all we use here

```suggestion
  const userId = useAppSelector((state) => state.user.id));
```

Reply:
Yeah makes sense, changed to fetch from the user slice now